src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -874,6 +894,11 @@ def set_vmss_extension(
 
     # pylint: disable=no-member
     version = _normalize_extension_version(publisher, extension_name, version, vmss.location)
+    extension_profile = vmss.virtual_machine_profile.extension_profile
+    if extension_profile:
+        extensions = extension_profile.extensions
+        if extensions:
+            extension_profile.extensions = [x for x in extensions if x.type.lower() != extension_name.lower() or x.publisher.lower() != publisher.lower()]  # pylint: disable=line-too-long
```

Comment:
VMSS still has only one extension. The difference are
1. For VMSS, the extension info is under `virtual_machine_profile.extension_profile`; while for `vm`, it is under `vm_instance.instance_view.extensions`
2. To reset the extension, for vmss we go through `vmss.update`; while for vm, it has a dedicated API called `client.virtual_machine_extensions.create_or_update`